conans/test/functional/package_id/transitive_header_only_test.py

```diff
@@ -0,0 +1,67 @@
+import unittest
+
+from conans.test.utils.tools import TestClient, GenConanfile
+
+
+class TransitiveIdsTest(unittest.TestCase):
+
+    def transitive_library_test(self):
+        # https://github.com/conan-io/conan/issues/6450
+        client = TestClient()
+        client.run("config set general.default_package_id_mode=full_version_mode")
+        client.save({"conanfile.py": GenConanfile()})
+        client.run("create . liba/1.0@")
+        client.run("create . liba/1.1@")
+        client.save({"conanfile.py": GenConanfile().with_require_plain("liba/1.0")})
+        client.run("create . libb/1.0@")
+        client.save({"conanfile.py": GenConanfile().with_require_plain("libb/1.0")})
+        client.run("create . libc/1.0@")
+        client.save({"conanfile.py": GenConanfile().with_require_plain("libc/1.0")
+                                                   .with_require_plain("liba/1.0")})
+        client.run("create . libd/1.0@")
+        # The consumer forces to depend on liba/2, instead of liba/1
+        client.save({"conanfile.py": GenConanfile().with_require_plain("libc/1.0")
+                                                   .with_require_plain("liba/1.1")})
+        client.run("create . libd/1.0@", assert_error=True)
+        # both B and C require a new binary
+        self.assertIn("liba/1.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache", client.out)
+        self.assertIn("libb/1.0:50b928d46d42051a461440161b017eb6d52e2dff - Missing", client.out)
+        self.assertIn("libc/1.0:4599994f77257f88443590022af4754d5520b753 - Missing", client.out)
```

Comment:
This shows that, if `libb` gets a different package_id, then `libc` gets a different package id too (and `libd` transitively). I would like to see that `libc` is affected by the change in `liba` while `libb` still computes the same packag_id. 

I would like to check **if (yes or no) transitive dependencies affect my package_id**, we all know that transitively a change in the package_id of one transitive dependency is affecting me because it is propagated down the graph modifying all the package_ids of the packages between us.

Reply:
This is what is proved in the other test. I added this here for completeness, I can remove it. But next test ``transitive_major_mode_test`` is clearly showing that.

Reply:
Ok, I'm a believer now. I'll remove the first test, I'm sure it's been tested somewhere else (probably the second one is also tested somewhere), and if you want you can make the dependency chain shorter (`A <- B (semver_major-A) <- C` can be enough)

---

But **we need to fix the blogpost**... and I need to change my mind (not sure if someone else's):

> Very important to note that transitive requirements (dependencies of my dependencies) are only encoded into my package ID through the package reference (Conan reference, package ID and revisions) of my requirements.

Reply:
About the first test, I have been looking for a while for a complete test that does this, without much success. As it is a very fast test (0.5 secs), better leave it.

The second test, the long chain is on purpose, I totally want to verify the chain 

```
E (Build) -> D (Missing/new ID) -> C (cached/same ID) -> B (Missing/new ID) -> A (cached/new created version)
```

Is possible, and you have both missing binaries both above and below C.

Would you manage to update the blog post, please? 
Created issue: https://github.com/conan-io/conan-io.github.io/issues/107